server/org.eclipse.emfcloud.ecore.glsp/src/main/java/org/eclipse/emfcloud/ecore/glsp/operationhandler/CreateClassifierNodeOperationHandler.java

```diff
@@ -55,15 +58,23 @@ public void executeOperation(CreateNodeOperation operation, GModelState modelSta
 		EClassifier eClassifier = createClassifier(operation.getElementTypeId());
 		setName(eClassifier, modelState);
 
-		EcoreFacade facade = EcoreModelState.getEcoreFacade(modelState);
-		facade.createShape(operation.getLocation());
+		Shape shape = EnotationFactory.eINSTANCE.createShape();
+		shape.setPosition(operation.getLocation().orElse(GraphUtil.point(0, 0)));
 
-		if (!modelAccess.addEClassifier(EcoreModelState.getModelState(modelState), eClassifier)) {
+		SemanticProxy proxy = EnotationFactory.eINSTANCE.createSemanticProxy();
+		proxy.setUri(getSemanticProxyUri(eClassifier));
+		shape.setSemanticElement(proxy);
+
+		if (!modelAccess.addEClassifier(EcoreModelState.getModelState(modelState), eClassifier, shape)) {
 			throw new GLSPServerException(
 					"Could not execute create operation on eClassifier: " + eClassifier.getName());
 		}
 	}
 
+	protected String getSemanticProxyUri(EClassifier eClassifier) {
+		return "//" + eClassifier.getName();
```

Comment:
Let's use URIs here as well, e.g.:

`EcoreUtil.getURI(eClassifier).fragment()`

Reply:
The `eClassifier` is not contained in any resource at this point of time, therefore we won't get a valid URI.

Reply:
Ah, I see. In that case, this is a good candidate for higher-level creation commands, where the GLSP Server should only request the creation of a Classifier Shape, and the Model Server should be responsible for creating the EClassifier, the Notation Shape, and create the Semantic Proxy. We can come back to this later.